File: backends/python-fastapi/app/usecase/gate/dto.py
"""
Gate ユースケース Dto モジュール。

Author: Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
"""
from dataclasses import dataclass


@dataclass
class GateIssueDto:
    """Gate JWT 発行 Dto。"""

    access_token: str
    member: str


@dataclass
class GateVerifyDto:
    """Gate JWT 検証 Dto。"""

    identifier: str
    token: str

File: backends/ts-hono/src/usecase/gate/dto.ts
/**
 * ゲートユースケース DTO モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** JWT 発行入力 DTO。 */
export interface GateIssueInput {
  accessToken: string;
  member: string;
}

/** JWT 検証入力 DTO。 */
export interface GateVerifyInput {
  identifier: string;
  token: string;
}
